### DISCUSSION TITLE
Welcome to bluesky Discussions!

### DISCUSSION THREAD

**Discussion by jooste** [TUDelft-CNS-ATM/bluesky#251] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Ellislee1** [comment]
Hi everyone, 👋
I'm Ellis a final year M.Sci Computer Science student in Swansea. My research interests started out with AI and Machine Learning. I was introduced bluesky from my undergraduate project, adapting a system by Marc Brittain to work with altitude.
This year I'm looking at how multi-agent reinforcement learning agents behave in sector-based environments and how sectors can be adapted for automated traffic management.

**Comment by jooste** [comment]
Hi Ellis! What a good discussion topic, multi-agent RL for BlueSky :) Have a look here: #253, let's see who else is working on this topic!

**Reply by Ellislee1** [reply]
Thanks @jooste! Its always exciting reading about work in the field!

**Comment by imanandpandey** [comment]
Hi Everyone,

I am an Air Traffic Controller from India. I have started learning python and Machine learning from last 10 months. I have used BlueSky desktop app. i want to contribute in its development. Can anyone suggest me where to start and a bit of briefing please?  Thanks in advance.

**Reply by jooste** [reply]
Hi!

Good to hear that you are eager to contribute to Bluesky. The most typical contributions to bluesky are researchers who implement a feature they need but bluesky does not have yet. So I guess it depends what you currently (want to) use it for!

**Reply by imanandpandey** [reply]
Hi,
Thanks for the reply Jooste. I am an OJTI at my airport. I want to use this simulator for simulating various traffic scenarios in my airspace for training my trainees. After that I want to use open source data to train bluesky to make it intelligent enough to predict a heading, departure delays, arrivals Expected Approach time which will help to reduce controllers workload.. I have been learning python and Machine learning for that.

**Reply by Amitoj-26** [reply]
Same i want to do

**Comment by davidwilcoxturbulence** [comment]
Hi Jooste,
I have just started learning BlueSky and was wondering if there was a way to get ADS-B data. I noticed that there is an opensky plugin in the code but I am not sure how to use it. Is there a documentation for this?
Many thanks

**Comment by jooste** [comment]
Hi @davidwilcoxturbulence,

Apologies for the delayed reply. There is unfortunately no documentation for any of the plugins yet, we'll have to put that on the TODO list :) 

You can load the plugin in BlueSky with the command `PLUGIN OPENSKY`. For it to work properly you need an account on OpenSky, and provide your login details in bluesky's `settings.cfg`:

```python
    opensky_user='username'
    opensky_password='mypassword'
```

Finally you can start/stop the feed with the command `OPENSKY ON/OFF`.

**Reply by davidwilcoxturbulence** [reply]
Thank you.

**Comment by davidwilcoxturbulence** [comment]
Hi, 
When I tried running the plugin mlcontrol, i get an AttributeError: 'Simulation object has no attribute 'send_event'.
I checked the simulation class and true enough, there isnt a function  named "send_event".
Can anybody help with this? Thank you :)

**Reply by jooste** [reply]
Hi David,

Ah, thanks, this is due to a change in the structure of bluesky that wasn't applied to this plugin. Have a look at commit 8c9430e, it should be fixed there!
